pytest-embedded/pytest_embedded/dut.py

```diff
@@ -80,11 +81,20 @@ def wrapper(
                         f'Please check the full log here: {self.logfile}'
                     )
                     raise e.__class__(debug_str) from e
+
+                if self.pexpect_proc.match in [pexpect.EOF, pexpect.TIMEOUT]:
+                    res.append(self.pexpect_proc.before.rstrip())
                 else:
-                    if self.pexpect_proc.match in [pexpect.EOF, pexpect.TIMEOUT]:
-                        res.append(self.pexpect_proc.before.rstrip())
-                    else:
-                        res.append(self.pexpect_proc.match)
+                    res.append(self.pexpect_proc.match)
+
+                for nm_pattern in to_list(not_matching):
+                    if isinstance(nm_pattern, str):
+                        nm_pattern = re.compile(nm_pattern.encode())
+                    if isinstance(nm_pattern.pattern, str):
+                        nm_pattern = re.compile(nm_pattern.pattern.encode())
```

Comment:
```suggestion
                    if isinstance(nm_pattern, re.Pattern):
                    	pass
```

otherwise it will be wrong when user is passing a `re.compile(b'xxx')`

Reply:
For me, isinstance(re.compile(b'hello').pattern, str) returns False. So, I think it will work as expected. This line is actually used to convert a string pattern to a byte-pattern.

Reply:
oh, yes. Please ignore this one :D